src/ts/Query.ts

```diff
@@ -0,0 +1,305 @@
+	}
+
+	/** Format the sort criteria object into a string to use in multivalue query */
+	private formatSortCriteria(criteria?: SortCriteria): string | null {
+		if (criteria == null || criteria.length === 0) {
+			return null;
+		}
+
+		return criteria
+			.map((sortEntry) => {
+				const [sortProperty, sortOrder] = sortEntry;
+				const dictionaryId = this.getDictionaryId(sortProperty);
+
+				let byClause;
+				if (sortOrder === -1) {
+					byClause = 'by.dsnd';
+				} else {
+					byClause = 'by';
+				}
+				return `${byClause} ${dictionaryId}`;
+			})
+			.join(' ');
+	}
+
+	/** Format a conditional expression */
+	private formatCondition(property: string, operator: string, value: unknown): string {
+		const dictionaryId = this.getDictionaryId(property);
+		return `${dictionaryId} ${operator} ${this.formatConstant(property, value)}`;
+	}
+
+	/**
+	 * Format a list of conditional expressions
+	 * @throws {@link InvalidParameterError} An invalid parameter was passed to the function
+	 */
+	private formatConditionList(
+		property: string,
+		operator: string,
+		valueList: unknown[],
+		joinString: string,
+	): string {
+		if (valueList.length === 0) {
+			throw new InvalidParameterError({ parameterName: 'valueList' });
+		}
+
+		const conditionList = valueList.map((value) => this.formatCondition(property, operator, value));
+
+		return conditionList.length === 1
+			? conditionList[0]
+			: `(${conditionList.join(` ${joinString} `)})`;
+	}
+
+	/**
+	 * Format a constant for use in queries
+	 * @throws {@link Error} Passed constant parameter contains both single and double quotes
+	 */
+	private formatConstant(property: string, constant: unknown): string {
+		const constantToFormat = this.transformToQuery(property, constant);
+
+		if (
+			typeof constantToFormat === 'string' &&
+			constantToFormat.includes(`'`) &&
+			constantToFormat.includes(`"`)
+		) {
+			// cannot query if string has both single and double quotes in it
+			throw new Error('Query constants cannot contain both single and double quotes');
+		}
+
+		const quoteCharacter =
+			typeof constantToFormat === 'string' && constantToFormat.includes(`"`) ? `'` : `"`;
+
+		return `${quoteCharacter}${constantToFormat}${quoteCharacter}`;
+	}
+
+	/**
+	 * Get a dictionary id at a given schema path
+	 * @throws {link InvalidParameterError} Nonexistent schema property or property does not have a dictionary specified
+	 */
+	private getDictionaryId(property: string): string {
+		const dictionaryId = this.Model.schema?.dictPaths[property];
+		if (dictionaryId == null) {
+			throw new InvalidParameterError({
+				message: 'Nonexistent schema property or property does not have a dictionary specified',
+				parameterName: 'property',
+			});
+		}
+		return dictionaryId;
+	}
+
+	/** Transform query constant to internal u2 format (if applicable) */
+	private transformToQuery(property: string, constant: unknown): unknown {
```

Comment:
This method did not exist previously.  A method which returned the transformer method was in its place.  Rather than having the caller get the method and then run it, it seemed better just to run the transform.